src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -314,4 +317,49 @@ public static void saveAndUpdateConfigs(
             throw ExceptionsHelper.convertToOpenSearchException(e);
         }
     }
+
+    /**
+     * Removes accounts that are not of the requested type from the SecurityDynamicConfiguration object passed.
+     *
+     * Accounts with the 'service' attribute set to true, are considered of type 'service'.
+     * Accounts with the 'service' attribute set to false or without the 'service' attribute, are considered of type 'internal'.
+     *
+     * @param configuration SecurityDynamicConfiguration object containing all accounts
+     * @param requestedAccountType The type of account to be kept. Should be "service" or "internal"
+     *
+     */
+    public void includeAccountsIfType(SecurityDynamicConfiguration<?> configuration, AccountType requestedAccountType)
+        throws UserServiceException {
+        if (requestedAccountType != AccountType.INTERNAL && requestedAccountType != AccountType.SERVICE) {
+            throw new UserServiceException("Invalid user type {} " + requestedAccountType);
+        }
+        List<String> toBeRemoved = new ArrayList<>();
+
+        if (requestedAccountType == AccountType.SERVICE) {
+            for (Map.Entry<String, ?> entry : configuration.getCEntries().entrySet()) {
+                final InternalUserV7 internalUserEntry = (InternalUserV7) entry.getValue();
+                final Map accountAttributes = internalUserEntry.getAttributes();
+                final String accountName = entry.getKey();
+                final boolean isServiceAccount = Boolean.parseBoolean(accountAttributes.getOrDefault("service", "false").toString());
+
+                if (isServiceAccount == false) {
+                    toBeRemoved.add(accountName);
+                }
+            }
+        }
+        else if (requestedAccountType == AccountType.INTERNAL) {
+
+            for (Map.Entry<String, ?> entry : configuration.getCEntries().entrySet()) {
+                final InternalUserV7 internalUserEntry = (InternalUserV7) entry.getValue();
+                final Map accountAttributes = internalUserEntry.getAttributes();
+                final String accountName = entry.getKey();
+                final boolean isServiceAccount = Boolean.parseBoolean(accountAttributes.getOrDefault("service", "false").toString());
+
+                if (isServiceAccount == true) {
+                    toBeRemoved.add(accountName);
+                }
+            }
+        }
+        configuration.remove(toBeRemoved);
```

Comment:
This method body has larges amounts of duplicate logic, lets see about improving it.  I had ChatGPT rewrite this code, but it can be condensed even more with logic abstracted which will make it easier to read and more obvious if there are bugs.

```
        final List<String> toBeRemoved = configuration.getCEntries().entrySet().stream()
            .filter(entry -> {
                InternalUserV7 internalUserEntry = (InternalUserV7) entry.getValue();
                Map<String, Object> accountAttributes = internalUserEntry.getAttributes();
                boolean isServiceAccount = Boolean.parseBoolean(accountAttributes.getOrDefault("service", "false").toString());
                return (requestedAccountType == AccountType.SERVICE && !isServiceAccount) ||
                       (requestedAccountType == AccountType.INTERNAL && isServiceAccount);
            })
            .map(Map.Entry::getKey)
            .collect(Collectors.toList());
        
        configuration.remove(toBeRemoved);
        ```

Reply:
I agree that code replication in general is not good, but the alternative in that case would be moving the 4 repeated lines into a new method, adding (imo) unnecessary complexity and a very 'meagre' method. 

The functional approach pasted above  has the same structure that I had done initially, with the RequestedAccountType being checked in every entry.

Can you confirm if we should move back to this initial approach?

Thanks

Reply:
With the most recent revision this is much more improved, good work! While I think it could still be even better - lets leave that to a future exercise if you are interested, but not block this review.  Progress not perfection!

Reply:
Thanks, but i didn't do anything xD, just signed-off the commit suggestion xD